packages/backend/src/rates/util.ts

```diff
@@ -11,9 +11,27 @@ export interface Asset {
   scale: number
 }
 
-export function convert(opts: ConvertOptions): bigint {
+export interface ConvertResults {
+  amount: bigint
+  scaledExchangeRate: number
+}
+
+export function convert(opts: ConvertOptions): ConvertResults {
   const scaleDiff = opts.destinationAsset.scale - opts.sourceAsset.scale
   const scaledExchangeRate = opts.exchangeRate * 10 ** scaleDiff
 
-  return BigInt(Math.round(Number(opts.sourceAmount) * scaledExchangeRate))
+  return {
+    amount: BigInt(Math.round(Number(opts.sourceAmount) * scaledExchangeRate)),
+    scaledExchangeRate
+  }
+}
+
+export function convertReverse(opts: ConvertOptions): ConvertResults {
+  const scaleDiff = opts.sourceAsset.scale - opts.destinationAsset.scale
+  const scaledExchangeRate = opts.exchangeRate * 10 ** scaleDiff
+
+  return {
+    amount: BigInt(Math.ceil(Number(opts.sourceAmount) / scaledExchangeRate)),
+    scaledExchangeRate
+  }
```

Comment:
This looks good and returing `scaledExchangeRate` makes a lot of sense, I'm wondering though whether `convertReverse` might be unclear. 


What do you think about the apporoach of having either `sourceAmount` OR `destinationAmount` provided as an argument to this function?
In our example, USD -> EUR, USD would be the `sourceAsset`, and when we pass in `destinationAmount` in EUR (without providing `sourceAmount`) the function would figure out out the proper direction.